provider/datagen/src/provider.rs

```diff
@@ -68,9 +68,10 @@ impl DatagenProvider {
             once_cell::sync::OnceCell::new();
         SINGLETON
             .get_or_init(|| {
+                let root = std::env::var("CARGO_MANIFEST_DIR")
+                    .expect("CARGO_MANIFEST_DIR must be set to run datagen tests");
```

Comment:
I think a good default would be to `unwrap_or_default()` if the variable isn't set, which will resolve to the current directory. If we want to be more compatible with non-cargo, we shouldn't force users to set that flag, instead they could run the tests in the appropriate location.

Reply:
That seems more prone to bugs IMO. I'd rather make that change when someone asks us to.

Reply:
What kind of bugs? I think requiring `CARGO_MANIFEST_DIR` is overly restricting. It probably works for us, but I don't see a need for this panic.

Reply:
Because this now depends on where you run `cargo test` from, and if you get that wrong you'll get a file not found error that will be annoying to debug.